app/controllers/file.rb

```diff
@@ -13,71 +13,72 @@ module Controllers
 		class File < Base
 			helpers Tint::Helpers::Rendering
 
-			namespace "/:site/files" do
-				get "/?*" do
-					file = site.file(params['splat'].join('/'))
+			set(:query) { |val| condition { request.query_string == val } }
 
-					if params[:download] && file.exist? && !file.directory?
-						authorize file, :show?
-						return send_file file.path, filename: file.name, type: file.mime, disposition: :attachment
+			namespace "/:site/files" do
+				get "/?*", query: "download" do
+					if resource.exist? && resource.file?
+						authorize resource, :show?
+						send_file resource.path, filename: resource.name, type: resource.mime, disposition: :attachment
 					end
+				end
 
-					if file.directory? || !file.exist?
-						authorize file.to_directory, :index?
-						render_directory file.to_directory
-					elsif file.text?
-						authorize file, :edit?
+				get "/?*" do
+					if resource.directory? || !resource.exist?
+						authorize resource, :index?
+						render_directory resource
+					elsif resource.text?
+						authorize resource, :edit?
 
 						if params.has_key?('source')
```

Comment:
this could use the `query` helper you made and be a different route block.  No real need to check if it's text or not if you specifically asked for source.